Add slurm

alexg9010 opened the pull request.

alexg9010 committed:
adding RSTUDIO_PANDOC to captured environment

alexg9010 committed:
use RSTUDIO_PANDOC on job submission

rekado commented:
This is a welcome addition.  I'd just like to point out that we really ought to be using DRMAA for job submission instead of handling each cluster type separately.

alexg9010 commented:
Thanks for this advice. I will have to look into DRMAA (from snakemake).

alexg9010 committed:
fix quoting of RSTUDIO_PANDOC to cluster jobs

alexg9010 committed:
add function to detect cluster system (SGE or SLURM) for job submission

alexg9010 committed:
add validation and conversion functions for cluster job arguments

alexg9010 committed:
introduce default cluster configuration with optional default SGE stack parameter

alexg9010 committed:
branch building of queue seletion string

alexg9010 committed:
branch building of contact mail string

alexg9010 committed:
pass the submission command during scheduler detection

alexg9010 committed:
pass ressources per scheduler

alexg9010 committed:
check and fix incompatible scheduler arguments

alexg9010 committed:
add more examples for scheduler argument conversion

alexg9010 committed:
add missing comma

alexg9010 committed:
fix slurm log file format

alexg9010 committed:
collate environment variables specfication

alexg9010 committed:
include custom cluster settings in cluster_config

alexg9010 committed:
export PIGX_UNINSTALLED env var

alexg9010 committed:
add parsable flag to sbatch

alexg9010 committed:
enable slurm mode for snakemake v7

alexg9010 committed:
allow dumping the submission command

alexg9010 committed:
allow custom cluster args per rule

alexg9010 committed:
add cancel command for slurm

alexg9010 committed:
remove the general slurm flag

- this does not transport custom args that we prepared for the cluster submission

alexg9010 committed:
safe retrieval of debug folder

alexg9010 committed:
ignore num_gpus key

alexg9010 committed:
add slurm-status script
